rmf_rviz_plugin/src/RmfPanel.cpp

```diff
@@ -13,404 +13,758 @@
+  _end_waypoint_selector = new QComboBox;
+  _end_waypoint_selector->setEditable(true);
+  selector_layout->addWidget(_end_waypoint_selector, 2, 1, 1, 2);
+
+  selector_layout->addWidget(new QLabel("Repeat Count: "), 3, 0);
+  _repeat_count_selector = new QSpinBox;
+  _repeat_count_selector->setValue(1);
+  selector_layout->addWidget(_repeat_count_selector, 3, 1, 1, 2);
+
+  selector_layout->addWidget(new QLabel("Time: "), 4, 0);
+  _time_selector = new QTimeEdit(QTime::currentTime());
+  _time_selector->setDisplayFormat(QString("hh:mm:ss ap"));
+  selector_layout->addWidget(_time_selector, 4, 1, 1, 2);
+
+  control_panel_layout->addWidget(selector_gb, 1, 0);
+
+  _update_time_checkbox = new QCheckBox("Keep Time Updated");
+  _update_time_checkbox->setChecked(true);
+  selector_layout->addWidget(_update_time_checkbox, 4, 3);
+
+  // Status
+  QGroupBox *status_gb = new QGroupBox("Status");
+  QGridLayout *status_layout = new QGridLayout();
+  status_gb->setLayout(status_layout);
+
+  _fleet_summary_view = new QListView;
+  status_layout->addWidget(_fleet_summary_view, 0, 0, 3, 5);
+
+  control_panel_layout->addWidget(status_gb, 2, 0);
+
+  // Plan
+  QGroupBox *plan_gb = new QGroupBox("Plan");
+  QGridLayout *plan_layout = new QGridLayout();
+  plan_gb->setLayout(plan_layout);
+
+  _pause_plan_checkbox = new QCheckBox("Pause Plan");
+  _pause_plan_checkbox->setChecked(false);
+  plan_layout->addWidget(_pause_plan_checkbox, 0, 1);
+
+  _plan_list_view = new QListView;
+  plan_layout->addWidget(_plan_list_view, 1, 0, 9, 2);
+
+  _load_action_plan_button = new QPushButton("Load");
+  plan_layout->addWidget(_load_action_plan_button, 11, 0, 1, 1);
+
+  _delete_plan_item_button = new QPushButton("Delete");
+  plan_layout->addWidget(_delete_plan_item_button, 11, 1, 1, 1);
+
+  control_panel_layout->addWidget(plan_gb, 3, 0);
+
+  // Actions
+  QGroupBox *actions_gb = new QGroupBox("Actions");
+  QGridLayout *actions_layout = new QGridLayout();
+  actions_gb->setLayout(actions_layout);
+
+  _send_delivery_button = new QPushButton("Send Delivery Request");
+  actions_layout->addWidget(_send_delivery_button, 1, 0, 1, 2);
+
+  _send_loop_button = new QPushButton("Send Loop Request");
+  actions_layout->addWidget(_send_loop_button, 2, 0, 1, 2);
+
+  _emergency_state_checkbox = new QCheckBox("Emergency State");
+  actions_layout->addWidget(_emergency_state_checkbox, 3, 1);
+
+  // Load File Dialog
+  _load_file_dialog = new QFileDialog();
+
+  control_panel_layout->addWidget(actions_gb, 4, 0);
 }
 
-RmfPanel::RmfPanel(QWidget* parent)
-: rviz_common::Panel(parent),
-  _delivery_task_id("task#42"),
-  _delivery_robot("magni"),
-  _delivery_pickup("pantry"),
-  _delivery_dropoff("hardware_2"),
-  _loop_task_id("task#24"),
-  _loop_robot("magni"),
-  _loop_start("coe"),
-  _loop_finish("cubicle_1"),
-  _loop_num("1")
+// Initialization Functions
+void RmfPanel::initialize_publishers(rclcpp::Node::SharedPtr _node) 
```

Comment:
there seems to be no need to pass the node as a parameter since its a protected member

Reply:
:+1: